Fix: Make some GSoC 2024-related changes (#7159)
jenkins-infra/jenkins.io: content/projects/gsoc/2024/project-ideas/cloudevents-plugin.adoc

---

=== Background

> As the CI/CD world is moving more towards interoperability between multiple platforms, Jenkins should also be compatible with the emergent interoperability standards. Some of these standards with respect to communication between different CI/CD platforms are put forth by the CloudEvents and CDEvents specifications. This spec outlines the structure of CloudEvents, which are produced or consumed by entities which support it, hence making those entities compatible with other CI/CD platforms which also support them allowing them to work together.

Reviewer: ```suggestion
As the CI/CD world moves towards interoperability between multiple platforms, Jenkins should also be compatible with the emergent interoperability standards.
Some of these standards concerning communication between different CI/CD platforms are put forth by the CloudEvents and CDEvents specifications.
This spec outlines the structure of CloudEvents, which are produced or consumed by entities that support it, hence making those entities compatible with other CI/CD platforms that allow them to work together.
```

> Jenkins currently supports CloudEvents via a link:../../2021/projects/cloudevents-plugin.adoc[former GSoC project from 2021], however, it does not support CDEvents, making it hard for users to use it with other platforms which support both.

Reviewer: ```suggestion
Jenkins currently supports CloudEvents via a link:../../2021/projects/cloudevents-plugin.adoc[former GSoC project from 2021], however, it does not support CDEvents, making it hard for users to use it with other platforms that support both.
```

The link:/projects/gsoc/2021/projects/cloudevents-plugin[previous CloudEvents plugin project] in Google Summer of Code 2021 was a predecessor to the CDEvents project, that is currently running in the Continuous Delivery Foundation, this is such that this older project could not support CDEvents since it existed before CDEvents' inception.
For CDEvents only, we have an link:https://plugins.jenkins.io/cdevents/[existing plugin] that is able to listen to and emit CDEvents, but does not support CloudEvents.
This project aims to extend Jenkins support to both the CloudEvents and CDEvents.
More information about that project is available at link:https://cdevents.dev/[].
For this new project, we will enable Jenkins to support CDEvents.
That may involve extending or adjusting the CloudEvents plugin so that it supports CDEvents.
The most likely scenario is that we will start a new plugin for this project.

> For a context, we expect the selected contributor to harness both the link:https://github.com/cloudevents/sdk-java[CloudEvents Java SDK] and the link:https://github.com/cdevents/sdk-java[CDEvents Java SDK] to create and read events by the proposed new plugin for Jenkins.
> The contributor will also need to understand how to Jenkins plugin development life-cycle works and gain hands-on experience in plugin architecture design.

Reviewer: ```suggestion
For context, we expect the selected contributor to harness both the link:https://github.com/cloudevents/sdk-java[CloudEvents Java SDK] and the link:https://github.com/cdevents/sdk-java[CDEvents Java SDK] to create and read events by the proposed new plugin for Jenkins.
The contributor must also understand how the Jenkins plugin development life-cycle works and gain hands-on experience in plugin architecture design.
```
